cmd/sealer/cmd/cluster/delete.go

```diff
@@ -118,3 +86,120 @@ func NewDeleteCmd() *cobra.Command {
 
 	return deleteCmd
 }
+
+func getRuntimeInterfaces(cf clusterfile.Interface) (imagedistributor.Interface, infradriver.InfraDriver, *clusterruntime.Installer, error) {
+	cluster := cf.GetCluster()
+	infraDriver, err := infradriver.NewInfraDriver(&cluster)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	runtimeConfig := new(clusterruntime.RuntimeConfig)
+	if cf.GetPlugins() != nil {
+		runtimeConfig.Plugins = cf.GetPlugins()
+	}
+
+	if cf.GetKubeadmConfig() != nil {
+		runtimeConfig.KubeadmConfig = *cf.GetKubeadmConfig()
+	}
+
+	installer, err := clusterruntime.NewInstaller(infraDriver, nil, *runtimeConfig)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	imageEngine, err := imageengine.NewImageEngine(imagecommon.EngineGlobalConfigurations{})
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	distributor, err := imagedistributor.NewScpDistributor(imageEngine, infraDriver)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	return distributor, infraDriver, installer, nil
+}
+
+func deleteCluster(workClusterfile string) error {
+	var err error
+	clusterFileData, err := ioutil.ReadFile(filepath.Clean(workClusterfile))
+	if err != nil {
+		return err
+	}
+	cf, err := clusterfile.NewClusterFile(clusterFileData)
+	if err != nil {
+		return err
+	}
+	distributor, infraDriver, installer, err := getRuntimeInterfaces(cf)
+	if err != nil {
+		return err
+	}
+	if err = installer.UnInstall(); err != nil {
+		return err
+	}
+	// exec clean.sh
+	ips := infraDriver.GetHostIPList()
+	clusterRootfsDir := infraDriver.GetClusterRootfs()
+	cleanFile := fmt.Sprintf(common.DefaultClusterClearBashFile, clusterRootfsDir)
+	for _, ip := range ips {
+		if err := infraDriver.CmdAsync(ip, cleanFile); err != nil {
+			return fmt.Errorf("failed to exec command(%s) on host(%s): error(%v)", cleanFile, ip, err)
+		}
+	}
+	//delete rootfs file
+	if err := distributor.Restore(infraDriver.GetClusterRootfs(), infraDriver.GetHostIPList()); err != nil {
+		return err
+	}
+
+	//todo delete CleanFs
+	if err := fs.FS.RemoveAll(common.GetClusterWorkDir(), common.DefaultClusterBaseDir(clusterName),
+		common.DefaultKubeConfigDir()); err != nil {
+		return err
+	}
+	return nil
+}
+
+func scaleDownCluster(workClusterfile string) error {
+	clusterFileData, err := ioutil.ReadFile(filepath.Clean(workClusterfile))
+	if err != nil {
+		return err
+	}
+	cf, err := clusterfile.NewClusterFile(clusterFileData)
+	if err != nil {
+		return err
+	}
+	cluster := cf.GetCluster()
+
+	hosts, err := utils.GetHosts(deleteArgs.Masters, deleteArgs.Nodes)
```

Comment:
1. do we need to validate input args : `mastersToDelete, workersToDelete`
2.  input args is not `deleteArgs.Masters, deleteArgs.Nodes`.